src/btree/bt_handle.c

```diff
@@ -549,8 +549,12 @@ __btree_conf(WT_SESSION_IMPL *session, WT_CKPT *ckpt)
     btree->syncing = WT_BTREE_SYNC_OFF; /* Not syncing */
                                         /* Checkpoint generation */
     btree->checkpoint_gen = __wt_gen(session, WT_GEN_CHECKPOINT);
+
+    /* Keep track of the latest write gen for this btree from the previous run. */
+    btree->base_write_gen = ckpt->write_gen;
```

Comment:
@keithbostic 
The crux of the change is here. Since the write gen is incremented per btree, I am maintaining a base write gen for each btree whereas the base write gen was previously the write gen of the metadata file (which is the highest write gen seen on any btree at the time of checkpoint).

If you imagine a case where one btree has gone through many write gens for whatever reason (more pages, more eviction, etc) and another btree which has gone through only a few write gens, the `base_write_gen` would be a very high number. And any pages written on the less active btree will have their transaction ids ignored with our previous strategy.

Reply:
I'm still confused here (yeah, you probably guessed that). Let me try to describe the two scenarios I see:
```
Generation 1
    Btree A: 1-50
    Btree B: 1-30
Generation 2-10
    Btree A: +50 more pages each generation

Btree A has generations 1-500 on disk, Btree B has generations 1-30 on disk.

The database restarts, and we start modifying both tables and write a few pages.
```
In model #1, Btree A has generations 1-510 on disk and Btree B has generations 1-30 and 501-510 on disk, and the test for Btree B is "generation less than 501 (the maximum base generation as of this restart)".

In model #2, Btree A has generations 1-510 on disk and Btree B has generations 1-40 on disk, and the test for Btree B is "generation less than 31 (the Btree generation for Btree B as of this restart)".

How are those two questions different?

Reply:
Ah, you're correct. I forgot the part where we start the write gen of all Btrees to the maximum at the time of restart:
```
btree->write_gen = WT_MAX(ckpt->write_gen, conn->base_write_gen);
```